mail_notification_custom_subject/models/mail_thread.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2020 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, api
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.multi
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(
+        self, body='', subject=None,
+        message_type='notification', subtype=None,
+        parent_id=False, attachments=None,
+        notif_layout=False, add_sign=True, model_description=False,
+        mail_auto_delete=True, **kwargs
+    ):
+        subtype_id = kwargs.get('subtype_id', False)
+        if not subtype_id:
+            subtype = subtype or 'mt_note'
+            if '.' not in subtype:
+                subtype = 'mail.%s' % subtype
+            subtype_id = self.env['ir.model.data'].xmlid_to_res_id(subtype)
```

Comment:
Don't fail if invalid:
```suggestion
            subtype_id = self.env['ir.model.data'].xmlid_to_res_id(subtype, raise_if_not_found=False)
        if subtype_id:
            ...
```